CC/src/DistanceComputationTools.cpp

```diff
@@ -2814,16 +2979,24 @@ ScalarType DistanceComputationTools::ComputeCloud2PlaneRobustMax(	GenericCloud*
     assert(cloud && planeEquation);
 	assert(percent < 1.0f);
 
+	if (!cloud)
+	{
+		return 0;
```

Comment:
Should these "`return 0`"s use the enum?

(This one and below.)

Reply:
yes 0 was intentional the various plane distance/error functions which return ScalarType are used in various places and I didn't want any unintended consequences. This should happen in 
```
computeCloud2PlaneDistanceRMS
ComputeCloud2PlaneRobustMax
ComputeCloud2PlaneMaxDistance
ComputeCloud2PlaneDistance
```

Reply:
Looking into some of where these get used I would support making changes to them in the future (to keep the PR reasonable).
here is one example of how these are used, this is from `fastMarchingForFacetExtraction::computeTCoefApprox`
```C++
ScalarType reprojError = CCLib::DistanceComputationTools::ComputeCloud2PlaneDistance(&Yk, theLSQPlaneEquation, m_errorMeasure);
			if (reprojError >= 0)
				return (1.0f - orientationConfidence) * static_cast<float>(reprojError); 
 
```
This looks to like a bug to me as both a perfect projection and an error will give the same result